float8_experimental/float8_aten_api.py

```diff
@@ -43,8 +63,18 @@ def mm_float8_emulated(
 lib.impl("mm_float8_emulated", mm_float8_emulated, "CPU")
 lib.impl("mm_float8_emulated", mm_float8_emulated, "CUDA")
 
+lib.define(
+    "cast_to_float8_tensor(Tensor x, Tensor x_scale, ScalarType float8_dtype, bool emulate) -> Tensor"
```

Comment:
You'll need to add `tags=torch.Tag.pt2_compliant_tag` to this `define()` function to mark the op as PT2 compliant (we will also need to audit it to actual promise that it is PT2 friendly)